src/traces/bar/set_positions.js

```diff
@@ -348,6 +358,11 @@ function setOffsetAndWidthInGroupMode(gd, pa, sieve) {
             // if width is an array, then clone it into t.barwidth.
             var newBarwidth = width.slice(0, calcTrace.length);
 
+            // guard against non-numeric items
+            for(j = 0; j < newBarwidth.length; j++) {
```

Comment:
@n-riesco is there a way to place this for-loop in a place common to `setOffsetAndWidthInGroupMode` and `setOffsetAndWidth` ?